src/main/java/de/tum/cit/aet/artemis/exercise/service/SubmissionFilterService.java

```diff
@@ -0,0 +1,138 @@
+package de.tum.cit.aet.artemis.exercise.service;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.cit.aet.artemis.exercise.domain.Submission;
+import de.tum.cit.aet.artemis.exercise.domain.participation.StudentParticipation;
+import de.tum.cit.aet.artemis.fileupload.domain.FileUploadSubmission;
+import de.tum.cit.aet.artemis.modeling.domain.ModelingSubmission;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingSubmission;
+import de.tum.cit.aet.artemis.quiz.domain.QuizExercise;
+import de.tum.cit.aet.artemis.quiz.domain.QuizSubmission;
+import de.tum.cit.aet.artemis.text.domain.TextSubmission;
+
+/**
+ * A service that filters the submissions of a participation mainly for the course dashboard. Different logic is
+ * required for the different types of exercises.
+ */
+@Service
+@Profile(PROFILE_CORE)
+public class SubmissionFilterService {
+
+    /**
+     * Filters a set of submissions from a participation and determines the latest submission with a valid result, based
+     * on the type of the exercise. This method is used to get data for the course dashboard, in which case the
+     * assessment due date matters, but also to get the latest results for each participation with points for each
+     * grading criterion, where the assessment due date can be ignored, because the corresponding endpoint is only
+     * visible for instructors.
```

Comment:
```suggestion
     * Filters a set of submissions from a participation and determines the latest submission with a valid result, based
     * on the type of the exercise.
     *
     * <p>This method can be used for example to get data for the course dashboard, in which case the
     * assessment due date matters, but also to get the latest results for each participation with points for each
     * grading criterion, where the assessment due date can be ignored, because the corresponding endpoint is only
     * visible for instructors.
```
nitpick: more used as examples when the `ignoreAssessmentDueDate` matters, rather than a definitive list ‘you can only used this method for those two exact things’.